### PR TITLE
Add Spotify Track Search to PlaylistPage

### PR DESCRIPTION
**Pull request by scottfirestone**

### CODE REVIEW

**Comment by scottfirestone** [comment]
@applegrain I'm trying to setState.searchResults within the callback function of the ajax request, but it seems to re-run the request and err. Can't figure out why this is happening. Any ideas?

**Comment by scottfirestone** [comment]
I initially called `setState` within the callback function (now commented out) and tried extracting it to another function. Same issue.

File: lib/components/PlaylistPage.js [hunk 1]
```diff
@@ -1,15 +1,84 @@
...
+
+    // this.state.searchResults.forEach(function(result){
+    //   console.log("1", result);
+    // });
+  },
+
+  storeResults(tracks) {
+    debugger;
+    self.state.searchResults = tracks;
+    self.setState({ searchResults : self.state.searchResults });
```
Comment (damwhit) [inline lib/components/PlaylistPage.js]: do you have access to self here?  Maybe try binding this to the store results function instead?

Comment (scottfirestone) [inline lib/components/PlaylistPage.js]: Good catch, definitely should have used `this` there. In the function making the AJAX call, I set `self` equal to `this` as `this` isn't scoped properly inside the AJAX callback.

I fixed that in my code and still have the same issue. I'll push up my changes.

File: lib/components/PlaylistPage.js [hunk 2]
```diff
@@ -1,15 +1,78 @@
...
+    e.preventDefault();
+    let queryValue = e.target.value;
+    this.searchSpotify(queryValue);
+  },
+
+  storeResults(tracks) {
+    debugger;
+    this.state.searchResults = tracks;
+    this.setState({ searchResults : self.state.searchResults });
+  },
```
Comment (amaxwellblair) [inline lib/components/PlaylistPage.js:29]: @damwhit commented on this earlier but I still think there could be some issues here:

you are attempting to set state in line 27 `this.state.searchResults = tracks;`
- This should not set state given you did not call `this.setState`
- Then in line 28 you attempt to take your newly set state and properly set state with it (I think this is where things go wrong)

One solution could be to delete line 27 & 28 and just set state once:
`this.setState({ searchResults : tracks });`

Comment (biglovisa) [inline lib/components/PlaylistPage.js:29]: yup, what Allan said! 

Let me know if that doesn't solve your problem @scottfirestone and I can take a closer look.

File: lib/components/SearchResults.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  if (results.length === 0) {
+    return <h1>No results yet!</h1>;
+  } else {
+    return (
+      <ul>{results.map((artist, track) =>
+        <li>
+          {artist}
+        </li>
+      )}</ul>
+    )
```
Comment (amaxwellblair) [inline lib/components/SearchResults.js:15]: if you replace lines 7-12 with `<div></div>` your double AJAX error should be resolved

**Comment by scottfirestone** [comment]
I need to add keys for each searchResult element but still having weird behavior with the ajax call, unsure if related to the 'no keys' error.

**Pull request merged**